docs/scos/dev/guidelines/performance-guidelines/spryker-elastic-computing/spryker-elastic-computing.md

```diff
@@ -0,0 +1,316 @@
+
+1. Update the `spryker/monitoring` module to version 2.5.0 or higher.
+2. In `Pyz\Zed\Monitoring\MonitoringConfig`, add `queue:task:start` command to the argument grouped transactions:
+```php
+namespace Pyz\Zed\Monitoring;
+
+use Spryker\Zed\Monitoring\MonitoringConfig as BaseMonitoringConfig;
+
+class MonitoringConfig extends BaseMonitoringConfig
+{
+    /**
+     * @api
+     *
+     * @return array<string>
+     */
+    public function getGroupedByArgumentTransactions(): array
+    {
+        return ['queue:task:start'];
+    }
+}
+```
+
+3. Add to the `Pyz\Zed\Monitoring\Business\MonitoringBusinessFactory` following lines of code:
+```php
+public function getMonitoringTransactionNamingStrategies(): array
+    {
+        return [
+            return new FirstArgumentMonitoringConsoleTransactionNamingStrategy($this->getConfig()->getGroupedByArgumentTransactions());
+        ];
+    }
+```
+This will aggregate all different queue worker transactions under corresponding `queue:task:start {queue_name}` transaction name in New Relic.
+
+As a result, the transactions will be displayed in the New Relic dashboardas follows.
+
+## RAM-aware batch processing integration
+
+Batch processing significantly speeds up processing operations. Moving more data into RAM at once decreases the number of I/O, thus winning some operations wall time. Dynamic batch size calcualtion based on available RAM wins even more time. 
+
+
+
+
+This can be achieved with elastic batch concept introduced in `spryker/data-import` module. The following instruction will explain the integration details of elastic batch using the glossary data import as an example.
+
+> :info: Similar integration must be applied for all project data importers.
+
+1. In `Pyz\Zed\DataImport\DataImportConfig`, add the following configuration:
+
+* `BulkWriteGradualityFactor`: estimates an upper limit of memory that can be safely utilized by operations. A bigger value enables a more precise approximation but requires more iterations. A smaller value gives a less precise approximation but requires less iterations.
+
+* `BulkWriteMemoryThresholdPercent`: defines a margin of PHP memory limit configured in PHP.
+
+
+```php
+/**
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ * @SuppressWarnings(PHPMD.ExcessiveMethodLength)
+ */
+class DataImportConfig extends SprykerDataImportConfig
+{
+    ...
+        
+    /**
+     * @return int
+     */
+    public function getBulkWriteGradualityFactor(): int
+    {
+        return 5;
+    }
+
+    /**
+     * @return int
+     */
+    public function getBulkWriteMemoryThesoldPercent(): int
+    {
+        return 95;
+    }
+}
+```
+
+2. In `src/Pyz/Zed/DataImport/Business/DataImportBusinessFactory.php`, adjust the creation of data importer so it’s new version will
+- use `DataSetStepBrokerElasticBatchTransactionAware` data set step broker and
```

Comment:
What is a `data set step broker`?

Reply:
This is a class (in our case) which is responsible for wrapping out a logic related to the processing of one data set (one row from the file with data).